CHANGELOG.md

```diff
@@ -59,6 +59,7 @@ Querying with using `{__mimir_storage__="ephemeral"}` selector no longer works.
 * [BUGFIX] Ingester: compact out-of-order data during `/ingester/flush` or when TSDB is idle. #4180
 * [BUGFIX] Ingester: conversion of global limits `max-series-per-user`, `max-series-per-metric`, `max-metadata-per-user` and `max-metadata-per-metric` into corresponding local limits now takes into account the number of ingesters in each zone. #4238
 * [BUGFIX] Ingester: track `cortex_ingester_memory_series` metric consistently with `cortex_ingester_memory_series_created_total` and `cortex_ingester_memory_series_removed_total`. #4312
+* [BUGFIX] Querier: fixed a bug which was incorrectly matching series with regular expression label matchers with begin/end anchors in the middle of the regular expression. #4340
```

Comment:
Also store-gateway, but not important I guess.

Reply:
Right. I think my take here is more like "a bug affecting queries, and any query go through the querier".

Reply:
:+1: